src/main/java/com/labsynch/labseer/chemclasses/rdkit/ChemStructureServiceRDKitImpl.java

```diff
@@ -268,11 +269,24 @@ public RDKitStructure getRDKitStructureFromService(String molfile) throws CmpdRe
 
 	@Override
 	public int saveStructure(String molfile, StructureType structureType, boolean checkForDupes) {
+
+		Long cdId=0L;
 		try {
 			RDKitStructure rdkitStructure = getRDKitStructureFromService(molfile);
 
 			if (structureType == StructureType.PARENT){
-				rdkitStructure.persist();
+				// Can't type cast from subclass to superclass so we go to json and back
+				RDKitStructure rdStructure = RDKitStructure.fromJsonToRDKitStructure(rdkitStructure.toJson());
+
+				if(checkForDupes){
+					List<RDKitStructure> rdkitStructures =  RDKitStructure.findRDKitStructuresByRegEquals(rdStructure.getReg()).getResultList();
+					if(rdkitStructures.size() > 0){
+						logger.error("RDkit parent structure already exists with id "+ rdkitStructures.get(0).getId());
+						return 0;
```

Comment:
This is an unexpected case, right? Since we expect the registration code above will catch this and not ask to `saveStructure`?

I think it would be safer to throw an exception rather than return 0, to make this break visibly in this unexpected state.

Reply:
I actually did try to find the right thing to do here and saw this:
https://github.com/mcneilco/acas-roo-server/blob/c942475947c904a7aa470828eb370449d0418f3b/src/main/java/com/labsynch/labseer/service/SaltFormServiceImpl.java#L160

And jchem does the same:
https://github.com/mcneilco/acas-roo-server/blob/7ecf6d6581af7286b55c0b5455ecff334db46f33/src/main/java/com/labsynch/labseer/chemclasses/jchem/ChemStructureServiceJChemImpl.java#L349

I am going to update the logger line above to be the following because its not always parent structure in this function:
```
						logger.error("RDkit structure for " + structureType + " type already exists with id "+ rdkitStructures.get(0).getId());
```